src/QsCompiler/CompilerOptimizations/Evaluation.fs

```diff
@@ -13,170 +13,181 @@ open Microsoft.Quantum.QsCompiler.SyntaxTree
 open Microsoft.Quantum.QsCompiler.Transformations.Core
 
 open ComputationExpressions
-open Types
 open Utils
 open Printer
 
 
+/// Represents the internal state of a function evaluation.
+/// The first element is a map that stores the current values of all the local variables.
+/// The second element is a counter that stores the remaining number of statements we evaluate.
+type private EvalState = Map<string, TypedExpression> * int
+
 /// Represents any interrupt to the normal control flow of a function evaluation.
 /// Includes return statements, errors, and (if they were added) break/continue statements.
 type private FunctionInterrupt =
 /// Represents the function invoking a return statement
 | Returned of TypedExpression
 /// Represents the function invoking a fail statement
 | Failed of TypedExpression
+/// We reached the limit of statement executions
+| OutOfStatements
```

Comment:
Pretty sure this is the opposite - you have more statements but reached the maximum evaluation count. ;)

Reply:
Would "TooManyStatements" be a better name?

Reply:
Fixed in latest commit